allennlp_rc/dataset_readers/transformer_squad.py

```diff
@@ -0,0 +1,325 @@
+        # Sequences are <s> question </s> </s> question </s>.
+        # This is technically not correct for anything but RoBERTa, but it does not affect the scores.
+        space_for_context = self.length_limit - len(tokenized_question) - 4
+        stride_start = 0
+        while True:
+            tokenized_context_window = tokenized_context[stride_start:]
+            tokenized_context_window = tokenized_context_window[:space_for_context]
+
+            window_token_answer_span = (
+                token_answer_span_start - stride_start,
+                token_answer_span_end - stride_start,
+            )
+            if any(i < 0 or i > len(tokenized_context_window) for i in window_token_answer_span):
+                # The answer is not contained in the window.
+                window_token_answer_span = None
+
+            if not self.skip_invalid_examples or window_token_answer_span is not None:
+                additional_metadata = {"id": qid}
+                instance = self.text_to_instance(
+                    question,
+                    tokenized_question,
+                    context,
+                    tokenized_context_window,
+                    answers,
+                    window_token_answer_span,
+                    additional_metadata,
+                )
+                yield instance
+
+            stride_start += space_for_context
+            if stride_start >= len(tokenized_context):
+                break
+            stride_start -= self.stride
+
+    @overrides
+    def text_to_instance(
+        self,  # type: ignore
+        question: str,
+        tokenized_question: List[Token],
+        context: str,
+        tokenized_context: List[Token],
+        answers: List[str],
+        token_answer_span: Optional[Tuple[int, int]],
+        additional_metadata: Dict[str, Any] = None,
+    ) -> Instance:
+        fields = {}
+
+        # make the question field
+        def cls_token():
+            return Token(
+                self._tokenizer.tokenizer.cls_token,
+                text_id=self._tokenizer.tokenizer.cls_token_id,
+                type_id=self.non_content_type_id,
+            )
+
+        def sep_token():
+            return Token(
+                self._tokenizer.tokenizer.sep_token,
+                text_id=self._tokenizer.tokenizer.sep_token_id,
+                type_id=self.non_content_type_id,
+            )
+
+        question_field = TextField(
+            (
+                [cls_token()]
+                + tokenized_question
+                + [sep_token(), sep_token()]
+                + tokenized_context
+                + [sep_token()]
+            ),
+            self._token_indexers,
+        )
+        fields["question"] = question_field
+        start_of_context = 1 + len(tokenized_question) + 2
+
+        # make the answer span
+        if token_answer_span is not None:
+            assert all(i >= 0 for i in token_answer_span)
```

Comment:
I'm personally not a fan of these asserts; it just distracts from the logic without giving you much benefit (this 14 line block could be 5 lines without the asserts).  Basically any problem that we see here would be caught in a test, except for a malformed data file, which seems pretty unlikely, and not the job of this code to test, anyway. (But, in case it's not obvious, just because I don't like them doesn't mean you have to remove them if you like them.)